client/shipping-label/state/actions.js

```diff
@@ -147,14 +147,28 @@ export const openPrintingFlow = () => ( dispatch, getState, context ) => {
 	let errors = getFormErrors( getState(), storeOptions );
 	const promisesQueue = [];
 
-	if ( ! origin.ignoreValidation && ! hasNonEmptyLeaves( errors.origin ) && ! origin.isNormalized && ! origin.normalizationInProgress ) {
+	if ( ! origin.ignoreValidation &&
+		! hasNonEmptyLeaves( errors.origin ) &&
+		! origin.isNormalized &&
+		! origin.normalizationInProgress ) {
 		promisesQueue.push( normalizeAddress( dispatch, origin.values, 'origin', addressNormalizationURL, nonce ) );
 	}
 
-	if ( ! destination.ignoreValidation && ! hasNonEmptyLeaves( errors.destination ) && ! destination.isNormalized && ! destination.normalizationInProgress ) {
+	if ( origin.ignoreValidation || hasNonEmptyLeaves( errors.origin ) ) {
+		dispatch( toggleStep( 'origin' ) );
+	}
+
+	if ( ! destination.ignoreValidation &&
+		! hasNonEmptyLeaves( errors.destination ) &&
+		! destination.isNormalized &&
+		! destination.normalizationInProgress ) {
 		promisesQueue.push( normalizeAddress( dispatch, destination.values, 'destination', addressNormalizationURL, nonce ) );
 	}
 
+	if ( origin.ignoreValidation || hasNonEmptyLeaves( errors.origin ) ) {
```

Comment:
Yes, it should :) My tests were also not catching this, so I fixed/expanded them, too.